src/components/pages/dashboard/panels/telemetry/telemetryPanel.js

```diff
@@ -0,0 +1,155 @@
+    super(props);
+
+    this.state = {
+      isPending: true,
+      telemetry: {},
+      telemetryKeys: [],
+      telemetryKey: ''
+    };
+
+    // Initialize chart client
+    this.tsiClient = new window.TsiClient();
+  }
+
+  componentDidMount() {
+    this.getData();
+
+    // Create line chart
+    this.lineChart = new this.tsiClient.ux.LineChart(document.getElementById(telemetryChartId));
+  }
+
+  componentWillUnmount() {
+    if (this.subscription) this.subscription.unsubscribe();
+  }
+
+  componentWillUpdate(_, { telemetry, telemetryKey}) {
+    if (telemetry && telemetryKey && telemetry[telemetryKey]) {
+      const datum = Object.keys(telemetry[telemetryKey]).map(deviceId => ({
+        [deviceId]: telemetry[telemetryKey][deviceId]
+      }));
+      const noAnimate = telemetryKey === this.state.telemetryKey;
+      // Set a timeout to allow the panel height to be calculated before updating the graph
+      setTimeout(() => {
+        this.lineChart.render(
+          datum,
+          {
+            grid: false,
+            legend: 'compact',
+            noAnimate, // If the telemetryKey changes, animate
+            tooltip: true,
+            yAxisState: 'shared' // Default to all values being on the same axis
+          },
+          chartColors.map(color => ({ color }))
+        );
+      }, 10);
+    }
+  }
+
+  getData() {
+    this.subscription = TelemetryService.getTelemetryByDeviceIdP15M()
+      .flatMap(response =>
+        Observable.interval(2000)
+          .flatMap(_ => TelemetryService.getTelemetryByDeviceIdP1M())
+          .startWith(response)
+      )
+      .flatMap(items =>
+        Observable.from(items)
+          .flatMap(({ data, deviceId, time }) =>
+            Observable.from(Object.keys(data))
+              .filter(key => key.indexOf('Unit') < 0)
+              .map(key => ({ key, deviceId, time, data: data[key] }))
+          )
+          .reduce((acc, { key, deviceId, time, data }) => ({
+            ...acc,
+            [key]: {
+              ...(acc[key] ? acc[key] : {}),
+              [deviceId]: {
+                ...(acc[key] && acc[key][deviceId] ? acc[key][deviceId] : {}),
+                '': {
+                  ...(acc[key] && acc[key][deviceId] && acc[key][deviceId][''] ? acc[key][deviceId][''] : {}),
+                  [time]: { val: data }
+                }
+              }
+            }
+          }), this.state.telemetry)
+      )
+      .subscribe(telemetry => {
+        const telemetryKeys = Object.keys(telemetry).sort();
+        this.setState({
+          telemetry,
+          telemetryKeys,
+          telemetryKey: this.state.telemetryKey || telemetryKeys[0],
+          isPending: false
+        });
+      }
+    );
+  }
+
+  setTelemetryKey = telemetryKey => () => this.setState({ telemetryKey });
+
+  render() {
+    const { telemetryKeys, telemetry, telemetryKey } = this.state;
+    return (
+      <Panel>
+        <PanelHeader>Telemetry</PanelHeader>
```

Comment:
Do we need to localize this text? If yes we should move it to a separate file

Reply:
Yes it does need to be localized. I haven't done so yet since this is the first pass. I'll do the localization pass when the functionality is working correctly. What do you mean by move to a separate file? You mean move the string? When the string is localized it will be moved to the `public/locales/en/translation.js` file.

Reply:
Yes moving to the locales folder is what i meant. It is ok to do it later

Reply:
If you are already tracking this then it is fine otherwise open a bug.